src/components/Api.js

```diff
@@ -0,0 +1,93 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this.baseUrl = baseUrl;
+    this.headers = headers;
+  }
+
+  // GET-LOAD the user's info and img from the server
+  getUserInfo() {
+    return fetch(`${this.baseUrl}/users/me`, {
+      headers: this.headers,
+    }).then(this._handleRequest);
+  }
+
+  //PATCH-EDIT the profile information (UPDATE)
+  updateProfile(name, about) {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this.headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).then(this._handleRequest);
+  }
+
+  //PATCH-EDIT the avatar in profile (UPDATE)
+  updateAvatar(avatarUrl) {
+    return fetch(`${this.baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this.headers,
+      body: JSON.stringify({
+        avatar: avatarUrl,
+      }),
+    }).then(this._handleRequest);
+  }
+
+  //GET-LOAD the initial card from the server
+  getInitialCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    }).then(this._handleRequest);
+  }
+
+  // POST-ADD a newCard name/link on to the server through createCard()
+  addNewCard(cardData) {
+    return fetch(`${this.baseUrl}/cards`, {
+      method: "POST",
+      headers: this.headers,
+      body: JSON.stringify(cardData),
+    })
+      .then((response) => response.json())
+      .then(this._handleRequests);
```

Comment:
Please, delete these 2 lines and use `.then(this._handleRequest);` here (without `s`)